Test/comp/Variance.dfy

```diff
@@ -0,0 +1,190 @@
+// RUN: %dafny /compile:3 /spillTargetCode:2 /compileTarget:cs "%s" > "%t"
+// RUN: %dafny /compile:3 /spillTargetCode:2 /compileTarget:js "%s" >> "%t"
+// RUN: %dafny /compile:3 /spillTargetCode:2 /compileTarget:go "%s" >> "%t"
+// RUN: %dafny /compile:3 /spillTargetCode:2 /compileTarget:java "%s" >> "%t"
+// RUN: %diff "%s.expect" "%t"
+// The Java compiler lacks support for this (see dafny0/RuntimeTypeTests0.dfy).
+
+datatype Co<+T> = Co(T){
+    const x : int;
```

Comment:
What's the purpose of these fields? Shouldn't they be used in some method?

Reply:
They, too, are exported through the interface, so I want to make sure that this works. They're used in the datatype, but also in the calling code (e.g. `Covariant`).